src/stylize.ts

```diff
@@ -0,0 +1,74 @@
+import {Channel} from 'vega-lite/src/channel';
+import {Type} from 'vega-lite/src/type';
+
+import {QueryConfig} from './config';
+import {SpecQueryModel} from './model';
+// import {Property} from './property';
+import {EncodingQuery, ScaleQuery} from './query/encoding';
+import {Schema} from './schema';
+import {Dict} from './util';
+
+export function stylize(answerSet: SpecQueryModel[], schema: Schema, opt: QueryConfig): SpecQueryModel[] {
+
+  answerSet = answerSet.map(function(specM) {
+    let encQIndex: Dict<EncodingQuery> = {};
+
+    if (opt.smallBandSizeForHighCardinality || opt.smallBandSizeForFacet) {
+
+      [Channel.ROW, Channel.Y, Channel.COLUMN, Channel.X].forEach((channel) => {
+        encQIndex[channel] = specM.getEncodingQueryByChannel(channel);
+      });
+
+      const yEncQ = encQIndex[Channel.Y];
+      if (yEncQ !== undefined) {
+        if (encQIndex[Channel.ROW] ||
+            schema.cardinality(yEncQ) > 10) {
+
+          if (yEncQ.scale === undefined) {
+            yEncQ.scale = yEncQ.scale || {};
```

Comment:
I don't understand why do you need to add
```
 if (yEncQ.scale === undefined) 
```

You either do 
```
if (yEncQ.scale === undefined) {
  yEncQ.scale = {};
}
```

or simply 
```
yEncQ.scale = yEncQ.scale || {};
```

which I prefer because it's shorter.   

The one you're writing doesn't make sense.